foxtrot-core/src/main/java/com/flipkart/foxtrot/core/table/impl/FoxtrotTableManager.java

```diff
@@ -0,0 +1,104 @@
+        try {
+            if (metadataManager.exists(table.getName())) {
+                throw new TableManagerException(TableManagerException.ErrorCode.BAD_REQUEST, "table already exists");
+            }
+            queryStore.init(table.getName());
+            dataStore.init(table);
+            metadataManager.save(table);
+        } catch (DataStoreException e) {
+            switch (e.getErrorCode()) {
+                case TABLE_NOT_FOUND:
+                    throw new TableManagerException(TableManagerException.ErrorCode.BAD_REQUEST, e);
+                case STORE_CONNECTION:
+                    throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+                default:
+                    throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+            }
+        } catch (TableManagerException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public Table get(String name) throws TableManagerException {
+        try {
+            Table table = metadataManager.get(name);
+            if (table == null) {
+                throw new TableManagerException(TableManagerException.ErrorCode.TABLE_NOT_FOUND, "table_does_not_exists");
+            }
+            return table;
+        } catch (TableManagerException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public List<Table> getAll() throws TableManagerException {
+        try {
+            return metadataManager.get();
+        } catch (Exception e) {
+            throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public void update(Table table) throws TableManagerException {
+        validateTableParams(table);
+        try {
+            if (!metadataManager.exists(table.getName())) {
+                throw new TableManagerException(TableManagerException.ErrorCode.TABLE_NOT_FOUND, "table_does_not_exists");
+            }
+            metadataManager.save(table);
+        } catch (TableManagerException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new TableManagerException(TableManagerException.ErrorCode.INTERNAL_ERROR, e);
+        }
+    }
+
+    @Override
+    public void delete(String tableName) throws TableManagerException {
+        // TODO Implement this once downstream implications are figured out
+    }
+
+    private void validateTableParams(Table table) throws TableManagerException {
+        if (table == null || table.getName() == null || table.getName().trim().isEmpty() || table.getTtl() <= 0) {
+            throw new TableManagerException(TableManagerException.ErrorCode.BAD_REQUEST, "Invalid Table Params");
```

Comment:
We still need to pass it through the validator itself in that case .. otherwise there will be two sources of checks.